parlai/core/torch_ranker_agent.py

```diff
@@ -567,6 +578,17 @@ def _build_candidates(self, batch, source, mode):
                         label_vec = label_vec[0: cand_vecs[i].size(1)]
                     label_vec_pad[0: label_vec.size(0)] = label_vec
                     label_inds[i] = self._find_match(cand_vecs, label_vec_pad)
+                    if label_inds[i] == -1:
+                        bad_batch = True
+                if bad_batch:
+                    if self.opt.get('ignore_bad_candidates'):
+                        label_inds = None
```

Comment:
this *may* break some things. hopefully this only happens when people are doing this intentionally. theoretically, it should only break training, so maybe we should have a check for `self.is_training` here too (i.e. `if self.opt.get('ignore_bad_candidates') and not self.is_training`)

Reply:
added, thanks